src/main/kotlin/org/elm/ide/toolwindow/ElmCompilerPanel.kt

```diff
@@ -0,0 +1,211 @@
+package org.elm.ide.toolwindow
+
+import com.intellij.icons.AllIcons
+import com.intellij.ide.CommonActionsManager
+import com.intellij.ide.OccurenceNavigator
+import com.intellij.ide.util.PsiNavigationSupport
+import com.intellij.openapi.Disposable
+import com.intellij.openapi.actionSystem.ActionManager
+import com.intellij.openapi.actionSystem.AnAction
+import com.intellij.openapi.actionSystem.CommonDataKeys
+import com.intellij.openapi.actionSystem.DefaultActionGroup
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.fileEditor.OpenFileDescriptor
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.ui.SimpleToolWindowPanel
+import com.intellij.openapi.vfs.LocalFileSystem
+import com.intellij.psi.PsiDocumentManager
+import com.intellij.psi.PsiManager
+import com.intellij.ui.AutoScrollToSourceHandler
+import com.intellij.ui.ColoredListCellRenderer
+import com.intellij.ui.JBSplitter
+import com.intellij.ui.SimpleTextAttributes.REGULAR_ATTRIBUTES
+import com.intellij.ui.components.JBList
+import com.intellij.ui.components.JBScrollPane
+import com.intellij.ui.content.ContentManager
+import org.elm.openapiext.checkIsEventDispatchThread
+import org.elm.openapiext.findFileByPath
+import org.elm.workspace.compiler.CompilerMessage
+import org.elm.workspace.compiler.ElmBuildAction
+import org.elm.workspace.compiler.Region
+import java.awt.Color
+import java.nio.file.Paths
+import javax.swing.JComponent
+import javax.swing.JList
+import javax.swing.JTextPane
+import javax.swing.ListSelectionModel
+
+
+class ElmCompilerPanel(private val project: Project, private val contentManager: ContentManager) : SimpleToolWindowPanel(true, false), Disposable, OccurenceNavigator {
+
+    private fun occurenceInfo(): OccurenceNavigator.OccurenceInfo {
+        var filePath = compilerMessages[indexCompilerMessages].path
+        if (!filePath.startsWith("/")) filePath = project.basePath + "/" + filePath // TODO any simpler way for this ?
+        val file = LocalFileSystem.getInstance().findFileByPath(Paths.get(filePath))
+        val psiFile = PsiManager.getInstance(project).findFile(file!!)
```

Comment:
it would be nice if we avoided `!!`. Does the caller expect a non-null return value? If not, just return null early.